pkg/asset/installconfig/gcp/validation.go

```diff
@@ -172,35 +170,80 @@ func validateManagedZones(client API, ic *types.InstallConfig, fieldPath *field.
 	return allErrs
 }
 
-// ValidatePreExitingPublicDNS ensure no pre-existing DNS record exists in the public
-// DNS zone for cluster's Kubernetes API.
-func ValidatePreExitingPublicDNS(client API, ic *types.InstallConfig) error {
+// ValidatePreExistingPrivateDNS ensure that the PrivateZone exists in the cluster domain
+func ValidatePreExistingPrivateDNS(client API, ic *types.InstallConfig) error {
+
+	if ic.Platform.GCP.PrivateDNSZone != nil && ic.Platform.GCP.PrivateDNSZone.ID != "" {
+		project := ic.Platform.GCP.PrivateDNSZone.ProjectID
+		if project == "" {
+			project = ic.Platform.GCP.ProjectID
+		}
+		returnedZone, err := client.GetDNSZoneByName(context.TODO(), project, ic.Platform.GCP.PrivateDNSZone.ID)
+		if err != nil {
+			if IsNotFound(err) {
+				return field.NotFound(field.NewPath("clusterDomain"), fmt.Sprintf("DNS Zone (%s/%s)", project, ic.ClusterDomain()))
+			}
+			if IsForbidden(err) {
+				errMsg := fmt.Sprintf("Permission denied. Unable to fetch private dns zone information: %s", ic.ClusterDomain())
+				return field.Invalid(field.NewPath("privateManagedZone"), ic.GCP.PrivateDNSZone.ID, errMsg)
+			}
+			return field.InternalError(field.NewPath("clusterDomain"), err)
+		}
+		if !strings.EqualFold(returnedZone.DnsName, ic.ClusterDomain()) {
+			return fmt.Errorf("dns zone %s did not match expected %s", returnedZone.DnsName, ic.ClusterDomain())
+		}
+	}
+
+	return nil
+}
+
+// ValidatePreExistingPublicDNS ensure no pre-existing DNS record exists in the public
+// DNS zone for cluster's Kubernetes API. If a PublicDNSZone is provided, the provided
+// zone is verified against the BaseDomain. If no zone is provided, the base domain is
+// checked for any public zone that can be used.
+func ValidatePreExistingPublicDNS(client API, ic *types.InstallConfig) error {
 	// If this is an internal cluster, this check is not necessary
 	if ic.Publish == types.InternalPublishingStrategy {
 		return nil
 	}
 
 	record := fmt.Sprintf("api.%s.", strings.TrimSuffix(ic.ClusterDomain(), "."))
+	zoneName := ""
```

Comment:
nit: this is technically a `zoneID` at this point as it cannot include a relative resource name.